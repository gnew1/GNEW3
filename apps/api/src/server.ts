import express, { type Application } from 'express';
import registerMissionsRoutes from './routes/missions.js';

const app: Application = express();
app.use(express.json({ limit: '1mb' }));

registerMissionsRoutes(app);

const port = process.env.PORT ? Number(process.env.PORT) : 3010;
app.listen(port, () => {
  console.log(`[api] listening on http://localhost:${port}`);
});
