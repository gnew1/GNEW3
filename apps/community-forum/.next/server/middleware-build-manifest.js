self.__BUILD_MANIFEST={polyfillFiles:["static/chunks/polyfills-78c92fac7aa8fdd8.js"],devFiles:[],ampDevFiles:[],lowPriorityFiles:[],rootMainFiles:[],pages:{"/":["static/chunks/webpack-8fa1640cc84ba8fe.js","static/chunks/framework-cc803994526a0d84.js","static/chunks/main-6b73ad10fdeb5643.js","static/chunks/268-63f933ec6a2c0dc3.js","static/chunks/pages/index-fa3c9f7a0fa2b415.js"],"/_app":["static/chunks/webpack-8fa1640cc84ba8fe.js","static/chunks/framework-cc803994526a0d84.js","static/chunks/main-6b73ad10fdeb5643.js","static/chunks/pages/_app-ca0f2fa5dc1cfd55.js"],"/_error":["static/chunks/webpack-8fa1640cc84ba8fe.js","static/chunks/framework-cc803994526a0d84.js","static/chunks/main-6b73ad10fdeb5643.js","static/chunks/pages/_error-1b42d646f7fed9b0.js"],"/new-thread":["static/chunks/webpack-8fa1640cc84ba8fe.js","static/chunks/framework-cc803994526a0d84.js","static/chunks/main-6b73ad10fdeb5643.js","static/chunks/pages/new-thread-093c6105daef9ea1.js"],"/threads":["static/chunks/webpack-8fa1640cc84ba8fe.js","static/chunks/framework-cc803994526a0d84.js","static/chunks/main-6b73ad10fdeb5643.js","static/chunks/604-f0672c0b191ab80a.js","static/chunks/268-63f933ec6a2c0dc3.js","static/chunks/pages/threads-a1982bb7633d094f.js"],"/threads/[id]":["static/chunks/webpack-8fa1640cc84ba8fe.js","static/chunks/framework-cc803994526a0d84.js","static/chunks/main-6b73ad10fdeb5643.js","static/chunks/604-f0672c0b191ab80a.js","static/chunks/pages/threads/[id]-d945dd2a4f4b686b.js"]},ampFirstPages:[]},self.__BUILD_MANIFEST.lowPriorityFiles=["/static/"+process.env.__NEXT_BUILD_ID+"/_buildManifest.js",,"/static/"+process.env.__NEXT_BUILD_ID+"/_ssgManifest.js"];