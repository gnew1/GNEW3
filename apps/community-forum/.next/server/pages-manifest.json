{"/":"pages/index.html","/new-thread":"pages/new-thread.html","/_app":"pages/_app.js","/api/threads/[id]":"pages/api/threads/[id].js","/api/threads":"pages/api/threads.js","/_error":"pages/_error.js","/threads/[id]":"pages/threads/[id].html","/threads":"pages/threads.html","/_document":"pages/_document.js","/404":"pages/404.html"}