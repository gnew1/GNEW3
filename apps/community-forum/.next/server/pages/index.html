<!DOCTYPE html><html><head><meta charSet="utf-8"/><meta name="viewport" content="width=device-width"/><meta name="next-head-count" content="2"/><noscript data-n-css=""></noscript><script defer="" nomodule="" src="/_next/static/chunks/polyfills-78c92fac7aa8fdd8.js"></script><script src="/_next/static/chunks/webpack-8fa1640cc84ba8fe.js" defer=""></script><script src="/_next/static/chunks/framework-8051a8b17472378c.js" defer=""></script><script src="/_next/static/chunks/main-42fe2c3d8a102521.js" defer=""></script><script src="/_next/static/chunks/pages/_app-5c4b95808db21a6a.js" defer=""></script><script src="/_next/static/chunks/139-ffaa35087cc93cbb.js" defer=""></script><script src="/_next/static/chunks/pages/index-2fa3a6dc84671c3e.js" defer=""></script><script src="/_next/static/oBY_7F9hCOYbbJFmfELfB/_buildManifest.js" defer=""></script><script src="/_next/static/oBY_7F9hCOYbbJFmfELfB/_ssgManifest.js" defer=""></script></head><body><div id="__next"><main class="max-w-3xl mx-auto py-12 px-4"><h1 class="text-3xl font-bold mb-6">GNEW Community Forum</h1><p class="mb-4">Discuss ideas, share feedback, and connect with other members.</p><ul class="list-disc ml-6 space-y-2"><li><a href="/threads">Threads</a></li><li><a href="/new-thread">Create New Thread</a></li></ul></main></div><script id="__NEXT_DATA__" type="application/json">{"props":{"pageProps":{}},"page":"/","query":{},"buildId":"oBY_7F9hCOYbbJFmfELfB","nextExport":true,"autoExport":true,"isFallback":false,"scriptLoader":[]}</script></body></html>