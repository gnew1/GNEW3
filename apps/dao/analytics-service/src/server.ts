
import express from "express";
import metricsRouter from "./api/metrics";

const app = express();

app.use("/metrics", metricsRouter);

const port = process.env.PORT || 4004;
app.listen(port, () => {
  console.log(`Analytics service running on port ${port}`);
});

Prompt N263 ‚Äî Microservicio de m√©tricas DAO

Stack: Node.js + Express + TypeScript + Prisma (PostgreSQL).

Crea un servicio de anal√≠ticas que recolecta y expone m√©tricas de:

Propuestas (totales, aprobadas, rechazadas, tasa de aprobaci√≥n).

Votaciones (total de votos, votantes √∫nicos, media de votos por propuesta).

Sistema (usuarios totales).

Expone endpoint GET /metrics.

üëâ Integrable en dashboards de administraci√≥n de la DAO GNEW.

