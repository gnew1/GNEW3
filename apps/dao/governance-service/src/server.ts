
import express from "express";
import votesRouter from "./api/votes";

const app = express();
app.use(express.json());

app.use("/votes", votesRouter);

const port = process.env.PORT || 4001;
app.listen(port, () => {
  console.log(`Governance service running on port ${port}`);
});

Prompt N246 — Implementación del microservicio de votaciones

Stack: Node.js + Express + TypeScript + Prisma (PostgreSQL).

Endpoints:

POST /votes crea o actualiza un voto (único por proposalId + voterId).

GET /votes/:proposalId devuelve todos los votos de una propuesta.

Prisma define las entidades Proposal y Vote con relación 1:N.

Este módulo se integra en el monorepo GNEW bajo apps/dao/governance-service.

