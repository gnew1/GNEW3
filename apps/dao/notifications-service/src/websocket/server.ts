
import { WebSocketServer, WebSocket } from "ws";
import http from "http";

type NotificationMessage = {
  type: "proposal_update" | "vote_cast" | "system_alert";
  payload: any;
};

const server = http.createServer();
const wss = new WebSocketServer({ server });

const clients = new Set<WebSocket>();

wss.on("connection", (ws: WebSocket) => {
  clients.add(ws);
  console.log("Client connected. Active clients:", clients.size);

  ws.on("message", (message) => {
    try {
      const parsed: NotificationMessage = JSON.parse(message.toString());
      console.log("Message from client:", parsed);
    } catch (err) {
      console.error("Invalid message", err);
    }
  });

  ws.on("close", () => {
    clients.delete(ws);
    console.log("Client disconnected. Active clients:", clients.size);
  });
});

export function broadcastNotification(notification: NotificationMessage) {
  const data = JSON.stringify(notification);
  clients.forEach((client) => {
    if (client.readyState === WebSocket.OPEN) {
      client.send(data);
    }
  });
}

const port = process.env.PORT || 4002;
server.listen(port, () => {
  console.log(`Notifications WebSocket service running on port ${port}`);
});


