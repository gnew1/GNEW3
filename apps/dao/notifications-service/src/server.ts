
import express from "express";
import notifyRouter from "./api/notify";
import "./websocket/server";

const app = express();
app.use(express.json());

app.use("/notify", notifyRouter);

const port = process.env.API_PORT || 4003;
app.listen(port, () => {
  console.log(`Notifications API running on port ${port}`);
});

Prompt N262 â€” Microservicio de notificaciones en tiempo real

Stack: Node.js + Express + TypeScript + WebSocket.

Implementa un servidor WebSocket para difundir notificaciones de propuestas y votos.

Expone un endpoint REST POST /notify que recibe un mensaje con { type, payload } y lo transmite a todos los clientes conectados.

Se ejecuta en el monorepo bajo apps/dao/notifications-service.

ðŸ‘‰ Listo para integrarse con el servicio de gobernanza y la UI de GNEW.

