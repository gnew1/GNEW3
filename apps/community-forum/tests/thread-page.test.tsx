import React from "react";
import { test, expect, vi } from "vitest";
import { renderToString } from "react-dom/server";
import ThreadPage from "../src/pages/threads/[id]";

vi.mock("next/router", () => ({
  useRouter: () => ({ query: { id: "1" } })
}));

vi.mock("swr", () => ({
  default: () => ({
    data: {
      title: "Test Thread",
      author: "Tester",
      posts: [
        { id: "msg-1", content: "First message", author: "Alice" },
        { id: "msg-2", content: "Second message", author: "Bob" }
      ]
    },
    error: null
  })
}));

test("thread page snapshot", () => {
  const html = renderToString(<ThreadPage />);
  expect(html).toMatchSnapshot();
});
