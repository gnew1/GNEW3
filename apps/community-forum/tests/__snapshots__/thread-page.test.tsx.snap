// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`thread page snapshot 1`] = `"<main class="max-w-2xl mx-auto p-4"><h1 class="text-2xl font-bold mb-4">Test Thread</h1><p class="text-sm text-gray-500">By <!-- -->Tester</p><div class="mt-4 space-y-2"><div class="border p-2 rounded"><p>First message</p><p class="text-xs text-gray-500">— <!-- -->Alice</p></div><div class="border p-2 rounded"><p>Second message</p><p class="text-xs text-gray-500">— <!-- -->Bob</p></div></div></main>"`;
